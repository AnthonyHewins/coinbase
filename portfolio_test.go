package coinbase

import (
	"bytes"
	"encoding/json"
	"fmt"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/shopspring/decimal"
)

func TestSpotJSON(mainTest *testing.T) {
	testCases := []struct {
		arg any
	}{
		{
			arg: SpotPosition{
				Asset:                "asset",
				AccountUUID:          "uuid",
				TotalBalanceFiat:     1,
				TotalBalanceCrypto:   2,
				AvailableToTradeFiat: 3,
				Allocation:           4,
				OneDayChange:         5,
				Cost_basis: Balance{
					Value:    decimal.NewFromInt(1),
					Currency: "2",
				},
				AssetImgUrl: "asdin",
				IsCash:      true,
			},
		},
	}

	for _, tc := range testCases {
		mainTest.Run(fmt.Sprintf("%T", tc.arg), func(t *testing.T) {
			b := bytes.NewBuffer([]byte{})
			m := json.NewEncoder(b)
			m.SetIndent("\t", "")
			err := m.Encode(tc.arg)
			if err != nil {
				t.Errorf("should not fail marshal")
				return
			}

			snaps.MatchSnapshot(t, b.String())
		})
	}
}
