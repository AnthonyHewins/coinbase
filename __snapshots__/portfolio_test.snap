
[TestSpotJSON/coinbase.SpotPosition - 1]
{
    "asset": "asset",
    "account_uuid": "uuid",
    "total_balance_fiat": 1,
    "total_balance_crypto": 2,
    "available_to_trade_fiat": 3,
    "allocation": 4,
    "one_day_change": 5,
    "cost_basis": {
    "currency": "2",
    "value": "1"
    },
    "asset_img_url": "asdin",
    "is_cash": true
    }

---
